### PR TITLE
Feature friendship

### PR DESCRIPTION
**Pull request by ccobasi**
## Second Milestone

- Create model
- Display all users
- Create association between user and friendship model
- Add friend request, accept request features

### CODE REVIEW

**Comment by mimipeshy** [comment]
## STATUS: CHANGES REQUESTED 🔴 ♻️ 

Hello there @ccobasi 👋🏾 this is a really strong first submission. 😄 You have used the correct gitflow and has integrated unit tests as well

However, before approval, let's work on the following ⬇️ 
- [x] Posts timeline includes posts written by logged-in user and their friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
- [x] User has to accept friendship request to start friendship ([screenshot ](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/accept_reject_friendship.png)from [Pathwright](https://microverse.pathwright.com/auth/sign-in/?next=/library/fast-track-curriculum/69047/path/step/49736080/))


That's all. 😄 , kindy submit another review when you are done

Regards
Peris Ndanu
Feel free to reach out on slack if you have any questions

File: app/views/friendships/_index.html.erb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<% @friendships.each do |friendship| %>
+    <tr>
+
+      <% @friendships = Friendship.where(user_id: current_user, status: :accept) %>
+    
+      <td><%= User.find(friendship.friend_id).name %></td>
+      
+    </tr>
+  <% end %>
```
Comment (mimipeshy) [inline app/views/friendships/_index.html.erb:9]: - [ ] There should be no logic in views, kindly make use of helpers

File: app/views/users/_user.html.erb [hunk 2]
```diff
@@ -1,6 +1,38 @@
...
+      <span>&nbsp</span>
+      <span>&nbsp</span>
+      <%= link_to 'Reject friend request',  update_request_path(user: current_user.id, friend: user, status: 'reject'), 
+      class: 'profile-link' %>
+    <% else %>
+      <%= link_to 'Invite friend',  friendships_path(user: @current_user, friend: user), :method => :post, 
+      class: 'profile-link' %>
+    <% end %>
+  <% end %>
+  </span>
```
Comment (mimipeshy) [inline app/views/users/_user.html.erb:37]: - [ ] There should be no logic in views, kindly use helpers

File: app/controllers/friendships_controller.rb [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  def reverse_frienship(data)
+    Friendship.create(user_id: data[:friend_id], friend_id: data[:user_id], status: data[:status])
+  end
+
+  def accept_or_reject_friend_request
+    friend_status = Friendship.where(friend_id: current_user.id, user_id: params[:friend])
+    friend_status.update(status: params[:status])
+
+    if params[:status] == 'accept'
+      # Friendship.create(user_id: params[:user], friend_id: params[:friend], status: params[:status])
```
Comment (mimipeshy) [inline app/controllers/friendships_controller.rb:37]: - [ ] Remove all commented out code

Comment (mimipeshy) [inline app/controllers/friendships_controller.rb]: - [x] Kindly remove commented out code

**Review by Temmarie — CHANGES_REQUESTED** [review]
## Changes Required! :recycle:

### Hi @ccobasi   ! :wave: :wave:

#### Thank you for implementing the previously required changes. However, I have to request changes to this project.
- - -

### **Recommended**
- [x] Posts timeline includes posts written by logged-in user and their friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

- The timeline should only have the login user's post and his or her friend's post so let's create these validations for the views.

- [x] User has to accept friendship request to start friendship ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/accept_reject_friendship.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

#### Test

- [x] I see you've created unit test cases with Rspec but let's also create integration test cases with Rspec and Capybara at least for the Friendship features (Send, Accept, and Reject a Friend Request and see user and friends post in the Timeline)

__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink:
__Grace Tamara Ekunola __
[GitHub](https://github.com/temmarie) | [LinkedIn](https://www.linkedin.com/in/ekunola-grace/) | [Twitter](https://twitter.com/TemmarieW)

**Review by GzimAsani — APPROVED** [review]
# STATUS: Millstone Accepted :partying_face: :green_circle: 

## Hello @ccobasi great job implementing the changes asked by previous TSE-s :clap: --- I have one optional change for you to make and it is very important for you to implement:
- [ OPTIONAL ] The invite friendship button in the user show page does not work so please improve that

_**PS: Don't forget to submit a project completion form**_

# HAPPY CODING :computer: :desktop_computer:

**Pull request merged**